src/pages/RFOX/components/Bridge/BridgeConfirm.tsx

```diff
@@ -0,0 +1,416 @@
+    ...reactQueries.common.evmFees({
+      value: '0',
+      accountNumber: accountNumber!, // see isGetApprovalFeesEnabled
+      feeAsset: feeAsset!, // see isGetApprovalFeesEnabled
+      feeAssetMarketData: feeAssetMarketData!, // see isGetApprovalFeesEnabled
+      to: fromAssetId(bridgeQuote.sellAssetId).assetReference,
+      from: fromAccountId(bridgeQuote.sellAssetAccountId).account,
+      data: approvalCallData!,
+      wallet: wallet!, // see isGetApprovalFeesEnabled
+    }),
+    staleTime: 30_000,
+    enabled: isGetApprovalFeesEnabled,
+    // Ensures fees are refetched at an interval, including when the app is in the background
+    refetchIntervalInBackground: true,
+    // Yeah this is arbitrary but come on, Arb is cheap
+    refetchInterval: isGetApprovalFeesEnabled ? 15_000 : false,
+  })
+
+  const serializedApprovalTxIndex = useMemo(() => {
+    if (!approvalTxHash) return ''
+    return serializeTxIndex(
+      bridgeQuote.sellAssetAccountId,
+      approvalTxHash,
+      fromAccountId(bridgeQuote.sellAssetAccountId).account,
+    )
+  }, [approvalTxHash, bridgeQuote.sellAssetAccountId])
+
+  const approvalTx = useAppSelector(gs => selectTxById(gs, serializedApprovalTxIndex))
+
+  const handleApprove = useCallback(() => sendApprovalTx(undefined), [sendApprovalTx])
+
+  const isApprovalTxPending = useMemo(
+    () =>
+      isApprovalMutationPending ||
+      (isApprovalMutationSuccess && approvalTx?.status !== TxStatus.Confirmed),
+    [approvalTx?.status, isApprovalMutationPending, isApprovalMutationSuccess],
+  )
+
+  const isApprovalTxSuccess = useMemo(
+    () => approvalTx?.status === TxStatus.Confirmed,
+    [approvalTx?.status],
+  )
+
+  // The approval Tx may be confirmed, but that's not enough to know we're ready to bridge
+  // Allowance then needs to be succesfully refetched - failure to wait for it will result in jumpy states between
+  // the time the Tx is confirmed, and the time the allowance is succesfully refetched
+  // This allows us to detect such transition state
+  const isTransitioning = useMemo(() => {
+    // If we don't have a success Tx, we know we're not transitioning
+    if (!isApprovalTxSuccess) return false
+    // We have a success approval Tx, but approval is still required, meaning we haven't re-rendered with the updated allowance just yet
+    if (isApprovalRequired) return true
+
+    // Allowance has been updated, we've finished transitioning
+    return false
+  }, [isApprovalRequired, isApprovalTxSuccess])
+
+  useEffect(() => {
+    if (!allowanceContract) return
+    if (!approvalTx) return
+    if (isApprovalTxPending) return
+    ;(async () => {
+      await queryClient.invalidateQueries(
+        reactQueries.common.allowanceCryptoBaseUnit(
+          bridgeQuote.sellAssetId,
+          allowanceContract,
+          fromAccountId(bridgeQuote.sellAssetAccountId).account,
+        ),
+      )
+    })()
```

Comment:
no need for IIFE with void keyword

```suggestion
    void queryClient.invalidateQueries(
        reactQueries.common.allowanceCryptoBaseUnit(
          bridgeQuote.sellAssetId,
          allowanceContract,
          fromAccountId(bridgeQuote.sellAssetAccountId).account,
        ),
      )
```

Reply:
But we probably don't even need the void keyword, we don't care about awaiting it or returning undefined isn't it?